public/app/features/alerting/components/AlertingQueryRows.tsx

```diff
@@ -0,0 +1,143 @@
+import React, { PureComponent } from 'react';
+import { DragDropContext, Droppable, DropResult } from 'react-beautiful-dnd';
+import { DataQuery, DataSourceApi, DataSourceInstanceSettings, rangeUtil, PanelData, TimeRange } from '@grafana/data';
+import { getDataSourceSrv } from '@grafana/runtime';
+import { QueryEditorRow } from 'app/features/query/components/QueryEditorRow';
+import { isExpressionQuery } from 'app/features/expressions/guards';
+import { GrafanaQuery } from 'app/types/unified-alerting-dto';
+
+interface Props {
+  // The query configuration
+  queries: GrafanaQuery[];
+
+  // Query editing
+  onQueriesChange: (queries: GrafanaQuery[]) => void;
+  onDuplicateQuery: (query: GrafanaQuery) => void;
+  onRunQueries: () => void;
+}
+
+interface State {
+  dataPerQuery: Record<string, PanelData>;
+  defaultDataSource: DataSourceApi;
+}
+
+export class AlertingQueryRows extends PureComponent<Props, State> {
+  constructor(props: Props) {
+    super(props);
+    this.state = { dataPerQuery: {}, defaultDataSource: {} as DataSourceApi };
+  }
+
+  async componentDidMount() {
+    const defaultDataSource = await getDataSourceSrv().get();
+    this.setState({ defaultDataSource });
```

Comment:
tiny nitpick and let's not waste time on it, but will mention anyway: `setState` will throw an error if component is unmounted before the `await` finishes